### ISSUE TITLE
borg recreate: wrong size of part files

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#3157]
from irc:
```
ganja\_$ When borg finds broken chunks like "var/lib/mysql/backup/mysqldump.sql.borg_part_1:
         size inconsistency detected: size 7546242262, chunks size 3184051881",
         wouldn't it be reasonable to be able to list all archives that are affected by that?
```

```
ThomasWaldmann$ do you have more details about the size inconsistency?
ganja\_$  Not except that it happened after a recreate and that check --repair isn't fixing it.
ganja\_$ And that it's consistent through multiple checks with several days between them.
       It's likely to be an effect from recreate since it has happened to 3 repos I ran
       recreate on (recompress).
```

**Comment by enkore** [comment]
Wouldn't a failure in this check either imply a Borg bug or a memory error?

**Comment by ThomasWaldmann** [comment]
Currently wondering about that a part file item has a size member?

**Comment by ThomasWaldmann** [comment]
from the fix changeset:
```
    borg recreate: correctly compute part file sizes, fixes #3157
    
    when doing in-file checkpointing, borg creates *.borg_part_N files.
    complete_file = part_1 + part_2 + ... + part_N
    
    the source item for recreate already has a precomputed (total) size
    member, thus we must force recomputation from the (partial) chunks
    list to correct the size to be the part's size only.
    
    borg create avoided this problem by computing the size member after
    writing all the parts. this is now not required any more.
    
    the bug is mostly cosmetic, borg check will complain, borg extract on
    a part file would also complain. but all the complaints only refer to
    the wrong metadata of the part files, the part files' contents are
    correct.
    
    usually you will never extract or look at part files, but only deal
    with the full file, which will be completely valid, all metadata and
    content.
    
    you can get rid of the archives with these cosmetic errors by running
    borg recreate on them with a fixed borg version. the old part files
    will get dropped (because they are usually ignored) and any new part
    file created due to checkpointing will be correct.
```
